src/MIDebugEngine/AD7.Impl/AD7BoundBreakpoint.cs

```diff
@@ -54,7 +51,22 @@ public AD7BoundBreakpoint(AD7Engine engine, AD7PendingBreakpoint pendingBreakpoi
         // Called when the breakpoint is being deleted by the user.
         int IDebugBoundBreakpoint2.Delete()
         {
-            return Delete();
+            if (Delete() == Constants.S_OK)
+            {
+                if ((enum_BP_LOCATION_TYPE)PendingBreakpoint.BpRequestInfo.bpLocation.bpLocationType == enum_BP_LOCATION_TYPE.BPLT_DATA_STRING)
+                {
+                    lock (_engine.DebuggedProcess.DataBreakpointVariables)
+                    {
+                        string addressName = HostMarshal.GetDataBreakpointStringForIntPtr(PendingBreakpoint.BpRequestInfo.bpLocation.unionmember3);
+                        if (_engine.DebuggedProcess.DataBreakpointVariables.Contains(addressName)) // might need to expand condition
```

Comment:
You shouldn't need to call .Contains. Firstly because removing a non-existing element doesn't throw, it just returns false. Secondly because it is probably a bug if the element isn't in the list, so we should probably be asserting.

Reply:
Would calling `.Contains` in the assert be okay?

Reply:
Sure